### DISCUSSION TITLE
dynamicNotifications functionality and static caching

### DISCUSSION THREAD

**Discussion by thisisjamessmith** [solspace/craft-freeform#518] [Questions & Answers] [answered]
Hiya,

It says in your docs about the dynamicNotifications parameter:

```
NOTE: this feature uses Session data. It will likely not work properly if the page is cached with something like Varnish, etc.
```
https://docs.solspace.com/craft/freeform/v4/template-functions/freeform.form.html#parameters

Could I ask for clarification of this? We're using static caching (Nginx FastCGI cache), and I have tested this functionality with the caching on, and as far as I can see it appears to be working fine.

We're doing the usual thing of dynamically refreshing the CSRF token and formHash inputs via AJAX for any FF forms on page load as per https://docs.solspace.com/craft/freeform/v4/templating/caching.html#static-page-caching-cdn-blitz - does this magically fix any issues we might have otherwise seen that are hinted at in that line in the docs?

**Comment by kjmartens** [comment]
Sorry for the delay @thisisjamessmith. It's possible some improvements to more recent Freeform 3 versions or Freeform 4 resolves this issue, but I'll check with a developer and get back to you. 🙂

**Comment by kjmartens** [comment, marked as answer]
Hi @thisisjamessmith,

This is a bit of dated information now, as it only applies when using the PHP Session. Ever since we added **Encrypted Payloads**, most people are using that (and it's the default), and it doesn't have this issue. I will update the docs. 🙂
